test/testdata/generate.sh

```diff
@@ -70,6 +70,12 @@ function scaffold_test_project {
     fi
 
     $kb create api --group crew --version v1 --kind Laker --controller=true --resource=false --make=false
+  elif [ $project == "project-v3-config-gen" ]; then
+    header_text 'Creating APIs ...'
+    $kb create api --group crew --version v1 --kind Captain --controller=true --resource=true --make=false
+    $kb create webhook --group crew --version v1 --kind Captain --defaulting --programmatic-validation
+    $kb create api --group crew --version v1 --kind Admiral --plural=admirales --controller=true --resource=true --namespaced=false --make=false
+    $kb create webhook --group crew --version v1 --kind Admiral --plural=admirales --defaulting
```

Comment:
Just to clarify, all the data under `testdata/project-v3-config-gen` is output by local `make generate`based on the update here. I am not sure if this is the correct way to generate test cases.

They pass the `e2e test` on my local laptop, but somehow failed in CI. It seems an issue of go version, not sure how I can fix that.

Reply:
The ci issue was not related. That is fixed now. 
Are we using the plugin go/v3 with? I think that we are not right
So, we might need to change the name of the dir to clarify that is the config-gen. 
If I am not wrong it has many samples right? if yes, should we create a dir like config-gen/ then add all samples inside? I will also need to look further on this one to have a better idea.

Reply:
IMO here we would add a sample that runs: 
`kubebuilder init --plugins=config-gen/v1-alpha,base.go.kubebuilder.io/v3 --domain example.org --repo example.org/guestbook-operator 

Such as we are doing in the PR where we propose the new kustomize/v2-alpha see: https://github.com/kubernetes-sigs/kubebuilder/pull/2583
`